app/Http/Controllers/Api/UsersController.php

```diff
@@ -74,6 +83,17 @@ public function store(UserRequest $request)
         }
 
         $contest->waitingRoom->users()->attach($user->id);
+        $this->manager->appendCampaign($contest);
+
+        // Fire off welcome Email
+        $message = Message::where(['contest_id' => $contest->id, 'type' => 'welcome'])->first();
```

Comment:
I think it'd be best as a separate function, as part of the `Message` class or `Email` class (whichever makes more sense), it could allow sending a Welcome message outside of the context of first signup or whatever. Would also help to keep things cleaner. Best to not do a lot in controller; separation of concerns.